worker/worker.go

```diff
@@ -290,15 +290,20 @@ func (w *Worker) handleError(queueMessage QueueMessage, msg *amqp.Delivery, err
 }
 
 func getNewMsg(msg *amqp.Delivery) *amqp.Delivery {
-	oldHeaderCount := msg.Headers["x-retry-count"]
-
-	if oldHeaderCount.(int32) >= maxRetries {
+	retryCount := int32(0)
+	if val, ok := msg.Headers["x-retry-count"].(int32); ok {
+		retryCount = val
+	}
+	if retryCount >= maxRetries {
 		return &amqp.Delivery{}
```

Comment:
Instead of returning empty struct it is better to return nil.

This way you will guarantee that returned value will not be used